src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/FakeTextBuffer.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Formatting
+{
+    internal class FakeTextBuffer : ITextBuffer
```

Comment:
Doesn't VS already have one of these publicly available? If not, we also [built one](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test.Common/TestTextBuffer.cs) back before a VS one existed. 

That being said, if this variant of the text buffer is super specific to formatting feel free to disregard this entirely.

Reply:
➡️ See https://github.com/dotnet/aspnetcore-tooling/pull/3061#discussion_r574821946